src/App.tsx

```diff
@@ -1,24 +1,202 @@
+    }
+  };
+
+  const clearCompletedTodos = () => {
+    const completedTodos = todos
+      .filter(({ completed }) => completed);
+
+    completedTodos.forEach(todo => {
+      handleTodoDelete(todo.id);
+    });
+  };
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setTodos(await getTodos(USER_ID));
+    };
+
+    fetchTodos();
+  }, []);
+
+  const handleFilterUpdate = (newFilter: FilterCases) => {
+    setFilterType(newFilter);
+  };
+
+  const handleNotificationClose = () => {
+    setError(ErrorTypes.None);
+  };
+
+  const handleTodoAdd = (data: TodoToSend) => {
+    const todo = addTodo(data);
+
+    setTempTodo({
+      ...data,
+      id: 0,
+    });
+
+    return todo;
+  };
+
+  const handleFormSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    let addedTodo: Todo;
+
+    if (!inputValue.trim()) {
+      generateError(ErrorTypes.EmptyTitleError);
+      setInputValue('');
+
+      return;
+    }
+
+    const newTodoData: TodoToSend = {
+      title: inputValue,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setIsAddingProceeding(true);
+    setInputValue('');
+
+    try {
+      addedTodo = await handleTodoAdd(newTodoData);
+
+      setTodos(current => [
+        ...current,
+        addedTodo,
+      ]);
+    } catch {
+      generateError(ErrorTypes.AddTodoError);
+    } finally {
+      setIsAddingProceeding(false);
+      setTempTodo(null);
+    }
+  };
+
+  const filteredArray = useMemo(() => {
+    return filterByStatus(todos, filterType);
+  }, [filterType, todos]);
+
+  const amountOfItemsLeft = todos
+    .filter(({ completed }) => !completed).length;
+
+  const isAllTodosActive = amountOfItemsLeft === 0;
+
+  const isTodosEmpty = todos.length === 0;
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {!isTodosEmpty && (
+            <ToggleButton isActive={isAllTodosActive} />
+          )}
+
+          <form
+            onSubmit={handleFormSubmit}
+          >
+            <input
+              disabled={isAddingProceeding}
+              value={inputValue}
+              onChange={(e) => setInputValue(e.target.value)}
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+            />
+          </form>
+        </header>
+
+        <section className="todoapp__main">
+          <TodoList
+            todos={filteredArray}
+            handleTodoDelete={handleTodoDelete}
```

Comment:
![image](https://user-images.githubusercontent.com/91411692/226335176-abea4bc8-208f-4f91-b4b1-af26c159e221.png)

```suggestion
            onDelete={handleTodoDelete}
```